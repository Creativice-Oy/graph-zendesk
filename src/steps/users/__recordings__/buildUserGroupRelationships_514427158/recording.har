{
  "log": {
    "_recordingName": "buildUserGroupRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "ed2fd5f5d5cd127ce3120f22e7ddec13",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "1"
            }
          ],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/groups.json?page=1"
        },
        "response": {
          "bodySize": 426,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 426,
            "text": "{\"groups\":[{\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json\",\"id\":360021738231,\"name\":\"Support\",\"description\":\"\",\"default\":true,\"deleted\":false,\"created_at\":\"2021-10-10T20:49:30Z\",\"updated_at\":\"2021-10-10T20:49:30Z\"}],\"next_page\":null,\"previous_page\":null,\"count\":1}"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:53:11 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "191"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "etag",
              "value": "W/\"76e3a2fd91ec6717583c8b3cf6682ee8\""
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-6rqbb"
            },
            {
              "name": "x-request-id",
              "value": "6d4b0c80299f2ca7-SEA, 6d4b0c80299f2ca7-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.128846"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=p0TiuKtknr%2BtiyOphj%2BJ13FlZK5lKlFhvuKjhF1Lobv%2FFsEflfKWGeUVxy6FSSYun%2B7TdR2O9UY0y%2F87yUly1i57teqNLKyJFLE4g9p09PkSXURpYqGkns8YYqWPI6ancMcJwQEVGDooBgUd\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4b0c80299f2ca7-ORD"
            }
          ],
          "headersSize": 1513,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:53:10.891Z",
        "time": 305,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 305
        }
      },
      {
        "_id": "1f7f1449eb5db9124d3e648a4715db0d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 329,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "1"
            }
          ],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/users.json?page=1"
        },
        "response": {
          "bodySize": 1728,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 1728,
            "text": "{\"users\":[{\"id\":420755677092,\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json\",\"name\":\"Austin Kelleher\",\"email\":\"development@jupiterone.dev\",\"created_at\":\"2021-10-10T20:49:22Z\",\"updated_at\":\"2022-01-28T14:53:10Z\",\"time_zone\":\"America/New_York\",\"iana_time_zone\":\"America/New_York\",\"phone\":null,\"shared_phone_number\":null,\"photo\":null,\"locale_id\":1,\"locale\":\"en-US\",\"organization_id\":361584776731,\"role\":\"admin\",\"verified\":true,\"external_id\":null,\"tags\":[],\"alias\":null,\"active\":true,\"shared\":false,\"shared_agent\":false,\"last_login_at\":\"2022-01-28T14:53:10Z\",\"two_factor_auth_enabled\":null,\"signature\":null,\"details\":null,\"notes\":null,\"role_type\":4,\"custom_role_id\":360019056171,\"moderator\":true,\"ticket_restriction\":null,\"only_private_comments\":false,\"restricted_agent\":false,\"suspended\":false,\"default_group_id\":360021738231,\"report_csv\":true,\"user_fields\":{}},{\"id\":420924998451,\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json\",\"name\":\"The Customer\",\"email\":\"customer@example.com\",\"created_at\":\"2021-10-10T20:49:34Z\",\"updated_at\":\"2021-10-10T20:49:36Z\",\"time_zone\":\"America/New_York\",\"iana_time_zone\":\"America/New_York\",\"phone\":null,\"shared_phone_number\":null,\"photo\":{\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460612.json\",\"id\":361209460612,\"file_name\":\"profile_image_420924998451_11522198.png\",\"content_url\":\"https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png\",\"mapped_content_url\":\"https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png\",\"content_type\":\"image/png\",\"size\":10254,\"width\":80,\"height\":80,\"inline\":false,\"deleted\":false,\"thumbnails\":[{\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460632.json\",\"id\":361209460632,\"file_name\":\"profile_image_420924998451_11522198_thumb.png\",\"content_url\":\"https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png\",\"mapped_content_url\":\"https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png\",\"content_type\":\"image/png\",\"size\":2402,\"width\":32,\"height\":32,\"inline\":false,\"deleted\":false}]},\"locale_id\":1,\"locale\":\"en-US\",\"organization_id\":null,\"role\":\"end-user\",\"verified\":false,\"external_id\":null,\"tags\":[],\"alias\":null,\"active\":true,\"shared\":false,\"shared_agent\":false,\"last_login_at\":null,\"two_factor_auth_enabled\":false,\"signature\":null,\"details\":null,\"notes\":null,\"role_type\":null,\"custom_role_id\":null,\"moderator\":false,\"ticket_restriction\":\"requested\",\"only_private_comments\":false,\"restricted_agent\":true,\"suspended\":false,\"default_group_id\":null,\"report_csv\":false,\"user_fields\":{}}],\"next_page\":null,\"previous_page\":null,\"count\":2}"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:53:11 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "etag",
              "value": "W/\"36c2112d81c32523e5f22107dba4e05b\""
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "188"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-fxdlr"
            },
            {
              "name": "x-request-id",
              "value": "6d4b0c8218052c9b-SEA, 6d4b0c8218052c9b-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.207737"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=0uWElMSEUKAsYvTgs54IcF3kv%2Bqsa%2B4UJ%2BshXk14At2TZy2DR%2FBV7ovADPnrRQWTlB3VTAbB8AOZiDZp1JW5ZTnN8zjvCMsEE%2Fi2ZOw0aOHeSR0HJoWKQnBY8kcr4SiaOKb6a%2FCzMJgpqZIk\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4b0c8218052c9b-ORD"
            }
          ],
          "headersSize": 1515,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:53:11.200Z",
        "time": 400,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 400
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
