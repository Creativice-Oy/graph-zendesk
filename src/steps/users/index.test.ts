import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { buildUserGroupRelationships, fetchUsers } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZendeskRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../config';
import { Entities, Relationships } from '../constants';
import { fetchGroups } from '../groups';

describe('#fetchUsers', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'fetchUsers',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes(Entities.USER._type),
    );

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: Entities.USER._class,
      schema: {
        additionalProperties: false,
        required: [],
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: Entities.USER._type },
          _class: { const: Entities.USER._class },
          _key: { type: 'string' },
          active: { type: 'boolean' },
          username: { type: 'string' },
          chatOnly: { type: 'boolean' },
          createdAt: { type: 'string' },
          customRoleId: { type: 'string' },
          defaultGroupId: { type: 'number' },
          email: { type: 'string' },
          ianaTimeZone: { type: 'string' },
          id: { type: 'string' },
          locale: { type: 'string' },
          localeId: { type: 'number' },
          moderator: { type: 'boolean' },
          name: { type: 'string' },
          notes: { type: 'array' },
          organizationId: { type: 'number' },
          phone: { type: 'string' },
          photoUrl: { type: 'string' },
          restrictedAgent: { type: 'boolean' },
          role: { type: 'string' },
          shared: { type: 'boolean' },
          suspended: { type: 'boolean' },
          timeZone: { type: 'string' },
          updatedAt: { type: 'string' },
          url: { type: 'string' },
          verified: { type: 'boolean' },
        },
      },
    });
  });
});

describe('#buildUserGroupRelationships', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'buildUserGroupRelationships',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchGroups(context);
    await fetchUsers(context);
    await buildUserGroupRelationships(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.GROUP_HAS_USER._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'zendesk_group_has_user',
          },
        },
      },
    });
  });
});
