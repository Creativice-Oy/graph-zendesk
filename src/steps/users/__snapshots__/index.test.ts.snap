// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#buildUserGroupRelationships should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "zendesk_group:360021738231",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2021-10-10T20:49:30Z",
            "default": true,
            "deleted": false,
            "description": "",
            "id": 360021738231,
            "name": "Support",
            "updated_at": "2021-10-10T20:49:30Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
          },
        },
      ],
      "_type": "zendesk_group",
      "createdAt": "2021-10-10T20:49:30Z",
      "createdOn": undefined,
      "deleted": false,
      "description": "",
      "displayName": "Support",
      "id": "360021738231",
      "name": "Support",
      "updatedAt": "2021-10-10T20:49:30Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420755677092",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:22Z",
            "custom_role_id": 360019056171,
            "default_group_id": 360021738231,
            "details": null,
            "email": "development@jupiterone.dev",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420755677092,
            "last_login_at": "2022-01-28T14:53:10Z",
            "locale": "en-US",
            "locale_id": 1,
            "moderator": true,
            "name": "Austin Kelleher",
            "notes": null,
            "only_private_comments": false,
            "organization_id": 361584776731,
            "phone": null,
            "photo": null,
            "report_csv": true,
            "restricted_agent": false,
            "role": "admin",
            "role_type": 4,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": null,
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": null,
            "updated_at": "2022-01-28T14:53:10Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
            "user_fields": Object {},
            "verified": true,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:22Z",
      "createdOn": undefined,
      "defaultGroupId": 360021738231,
      "displayName": "Austin Kelleher",
      "email": "development@jupiterone.dev",
      "ianaTimeZone": "America/New_York",
      "id": "420755677092",
      "locale": "en-US",
      "localeId": 1,
      "moderator": true,
      "name": "Austin Kelleher",
      "organizationId": 361584776731,
      "phone": undefined,
      "photoUrl": undefined,
      "restrictedAgent": false,
      "role": "admin",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2022-01-28T14:53:10Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
      "username": "development@jupiterone.dev",
      "verified": true,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420924998451",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:34Z",
            "custom_role_id": null,
            "default_group_id": null,
            "details": null,
            "email": "customer@example.com",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420924998451,
            "last_login_at": null,
            "locale": "en-US",
            "locale_id": 1,
            "moderator": false,
            "name": "The Customer",
            "notes": null,
            "only_private_comments": false,
            "organization_id": null,
            "phone": null,
            "photo": Object {
              "content_type": "image/png",
              "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "deleted": false,
              "file_name": "profile_image_420924998451_11522198.png",
              "height": 80,
              "id": 361209460612,
              "inline": false,
              "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "size": 10254,
              "thumbnails": Array [
                Object {
                  "content_type": "image/png",
                  "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "deleted": false,
                  "file_name": "profile_image_420924998451_11522198_thumb.png",
                  "height": 32,
                  "id": 361209460632,
                  "inline": false,
                  "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "size": 2402,
                  "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460632.json",
                  "width": 32,
                },
              ],
              "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460612.json",
              "width": 80,
            },
            "report_csv": false,
            "restricted_agent": true,
            "role": "end-user",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": "requested",
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": false,
            "updated_at": "2021-10-10T20:49:36Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
            "user_fields": Object {},
            "verified": false,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "defaultGroupId": undefined,
      "displayName": "The Customer",
      "email": "customer@example.com",
      "ianaTimeZone": "America/New_York",
      "id": "420924998451",
      "locale": "en-US",
      "localeId": 1,
      "moderator": false,
      "name": "The Customer",
      "organizationId": undefined,
      "phone": undefined,
      "photoUrl": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
      "restrictedAgent": true,
      "role": "end-user",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-10-10T20:49:36Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
      "username": "customer@example.com",
      "verified": false,
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "zendesk_group:360021738231",
      "_key": "zendesk_group:360021738231|has|zendesk_user:420755677092",
      "_toEntityKey": "zendesk_user:420755677092",
      "_type": "zendesk_group_has_user",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "zendesk_group",
    "zendesk_user",
    "zendesk_group_has_user",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 1,
}
`;

exports[`#fetchUsers should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420755677092",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:22Z",
            "custom_role_id": 360019056171,
            "default_group_id": 360021738231,
            "details": null,
            "email": "development@jupiterone.dev",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420755677092,
            "last_login_at": "2022-01-28T14:53:10Z",
            "locale": "en-US",
            "locale_id": 1,
            "moderator": true,
            "name": "Austin Kelleher",
            "notes": null,
            "only_private_comments": false,
            "organization_id": 361584776731,
            "phone": null,
            "photo": null,
            "report_csv": true,
            "restricted_agent": false,
            "role": "admin",
            "role_type": 4,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": null,
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": null,
            "updated_at": "2022-01-28T14:53:10Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
            "user_fields": Object {},
            "verified": true,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:22Z",
      "createdOn": undefined,
      "defaultGroupId": 360021738231,
      "displayName": "Austin Kelleher",
      "email": "development@jupiterone.dev",
      "ianaTimeZone": "America/New_York",
      "id": "420755677092",
      "locale": "en-US",
      "localeId": 1,
      "moderator": true,
      "name": "Austin Kelleher",
      "organizationId": 361584776731,
      "phone": undefined,
      "photoUrl": undefined,
      "restrictedAgent": false,
      "role": "admin",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2022-01-28T14:53:10Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
      "username": "development@jupiterone.dev",
      "verified": true,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420924998451",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:34Z",
            "custom_role_id": null,
            "default_group_id": null,
            "details": null,
            "email": "customer@example.com",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420924998451,
            "last_login_at": null,
            "locale": "en-US",
            "locale_id": 1,
            "moderator": false,
            "name": "The Customer",
            "notes": null,
            "only_private_comments": false,
            "organization_id": null,
            "phone": null,
            "photo": Object {
              "content_type": "image/png",
              "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "deleted": false,
              "file_name": "profile_image_420924998451_11522198.png",
              "height": 80,
              "id": 361209460612,
              "inline": false,
              "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "size": 10254,
              "thumbnails": Array [
                Object {
                  "content_type": "image/png",
                  "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "deleted": false,
                  "file_name": "profile_image_420924998451_11522198_thumb.png",
                  "height": 32,
                  "id": 361209460632,
                  "inline": false,
                  "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "size": 2402,
                  "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460632.json",
                  "width": 32,
                },
              ],
              "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460612.json",
              "width": 80,
            },
            "report_csv": false,
            "restricted_agent": true,
            "role": "end-user",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": "requested",
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": false,
            "updated_at": "2021-10-10T20:49:36Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
            "user_fields": Object {},
            "verified": false,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "defaultGroupId": undefined,
      "displayName": "The Customer",
      "email": "customer@example.com",
      "ianaTimeZone": "America/New_York",
      "id": "420924998451",
      "locale": "en-US",
      "localeId": 1,
      "moderator": false,
      "name": "The Customer",
      "organizationId": undefined,
      "phone": undefined,
      "photoUrl": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
      "restrictedAgent": true,
      "role": "end-user",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-10-10T20:49:36Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
      "username": "customer@example.com",
      "verified": false,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "zendesk_user",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 0,
}
`;
